backend/character/character.py

```diff
@@ -0,0 +1,120 @@
+''' Representation of a single character and its stat spread '''
+
+import math
+from dataclasses import dataclass
+from backend.character.stat import Stat, Stats, ProbabalisticStat
+from backend.character.jobs import Roles, Buffs
+
+@dataclass
+class CharacterStatSpread: #pylint: disable=too-many-instance-attributes
+    """
+    Value class representation of a stat spread
+    """
+    wd: float #pylint: disable=invalid-name
+    mainstat: int
+    det: int
+    crit: int
+    dh: int #pylint: disable=invalid-name
+    speed: int
+    ten: int
+    pie: int
+
+class Character:
+    """
+    The main class where damage is calculated. Initialized by providing a CharacterStatSpread.
+    """
+    def __init__(self, job, stat_spread):
+        self.job = job
+        self.wd = stat_spread.wd  #pylint: disable=invalid-name
+        self.stats = {}
+        self.stats[Stats.MAINSTAT] = Stat(Stats.MAINSTAT, stat_spread.mainstat)
+        self.stats[Stats.DET] = Stat(Stats.DET, stat_spread.det)
+        self.stats[Stats.CRIT] = ProbabalisticStat(Stats.CRIT, stat_spread.crit)
+        self.stats[Stats.DH] =  ProbabalisticStat(Stats.DH, stat_spread.dh)
+        self.stats[Stats.SPEED] = Stat(Stats.SPEED, stat_spread.speed)
+        self.stats[Stats.TEN] = Stat(Stats.TEN, stat_spread.ten)
+        self.stats[Stats.PIE] = Stat(Stats.PIE, stat_spread.pie)
+
+    def get_gcd(self):
+        '''
+        Returns the character's gcd given its skill or spell speed
+        :return: the gcd in seconds
```

Comment:
i think these are reST docstrings? so `:return:` should be `:returns:` and again all docstrings should use double, not single quotes